src/slices/userSlice.test.ts

```diff
@@ -0,0 +1,84 @@
+import * as userSlice from "./userSlice";
+import { AnyAction } from "redux";
+import { testSaga, expectSaga } from "redux-saga-test-plan";
+import * as matchers from "redux-saga-test-plan/matchers";
+import { throwError } from "redux-saga-test-plan/providers";
+
+const sampleData: userSlice.User = require("./__mocks__/sampleUser.json");
+
+describe("user slice", () => {
+  describe("combined reducer", () => {
+    it("returns the expected default initial state", () => {
+      const reducer = userSlice.slice.reducer;
+      const initialState = reducer(undefined, {} as AnyAction);
+      expect(initialState).toMatchSnapshot();
+    });
+
+    it("handles requestUser action", () => {
+      const reducer = userSlice.slice.reducer;
+      const actionCreator = userSlice.actions.requestUser;
+      expect(reducer(undefined, actionCreator())).toMatchSnapshot();
+    });
+
+    it("handles receiveUserSuccess action", () => {
+      const reducer = userSlice.slice.reducer;
+      const actionCreator = userSlice.actions.receiveUserSuccess;
+      expect(reducer(undefined, actionCreator(sampleData))).toMatchSnapshot();
+    });
+
+    it("handles receiveUserFailure action", () => {
+      const reducer = userSlice.slice.reducer;
+      const actionCreator = userSlice.actions.receiveUserFailure;
+      const error = new Error("Sample error message");
+      expect(reducer(undefined, actionCreator(error))).toMatchSnapshot();
+    });
+  });

```

Comment:
Do we need to test the reducer explicitly, when the saga tests are testing it implicitly?